tests/Meziantou.Analyzer.Test/Rules/AvoidComparisonWithBoolConstantAnalyzerTests.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System.Threading.Tasks;
+using Meziantou.Analyzer.Rules;
+using TestHelper;
+using Xunit;
+
+namespace Meziantou.Analyzer.Test.Rules
+{
+    public sealed class AvoidComparisonWithBoolConstantAnalyzerTests
+    {
+        private static ProjectBuilder CreateProjectBuilder()
+        {
+            return new ProjectBuilder()
+                .WithAnalyzer<AvoidComparisonWithBoolConstantAnalyzer>()
+                .WithCodeFixProvider<AvoidComparisonWithBoolConstantFixer>();
+        }
+
+        [Theory]
+        [InlineData("==", "true", "")]
+        [InlineData("==", "false", "!")]
+        [InlineData("!=", "true", "!")]
+        [InlineData("!=", "false", "")]
+        public async Task ComparingVariableWithBoolLiteral_KeepsVariable(string equalityOperator, string boolLiteral, string expectedPrefix)
+        {
+            var originalCode = $@"
+class TestClass
+{{
+    void Test()
+    {{
+        var condition = false;
+        if (condition [|{equalityOperator}|] {boolLiteral})
+        {{
+        }}
+    }}
+}}";
+            var modifiedCode = $@"
+class TestClass
+{{
+    void Test()
+    {{
+        var condition = false;
+        if ({expectedPrefix}condition)
+        {{
+        }}
+    }}
+}}";
+            await CreateProjectBuilder()
+                  .WithSourceCode(originalCode)
+                  .ShouldFixCodeWith(modifiedCode)
+                  .ValidateAsync();
+        }
+
+        [Theory]
+        [InlineData("true", "==", "")]
+        [InlineData("false", "==", "!")]
+        [InlineData("true", "!=", "!")]
+        [InlineData("false", "!=", "")]
+        public async Task ComparingBoolLiteralWithExpression_KeepsExpression(string boolLiteral, string equalityOperator, string expectedPrefix)
+        {
+            var originalCode = $@"
+class TestClass
+{{
+    void Test()
+    {{
+        var success = {boolLiteral} [|{equalityOperator}|] (GetSomeNumber() == 15);
+        int GetSomeNumber() => 12;
+    }}
+}}";
+            var modifiedCode = $@"
+class TestClass
+{{
+    void Test()
+    {{
+        var success = {expectedPrefix}(GetSomeNumber() == 15);
```

Comment:
In the case `var success = true == (GetSomeNumber() == 15);`, the fix should remove the parentheses: `var success = GetSomeNumber() == 15;`.

I think you need to add the `Simplifier` annotation